packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1419,6 +1422,24 @@ abstract class BuildableElement extends Element {
     markNeedsBuild();
   }
 
+  bool _hadDependenciesWhenDeactivated = false;
```

Comment:
That's tricky. The problem is that we have to keep track of whether we were added to the dirty list, and that's what _dirty usually tracks. So if we also use it to track whether we need to be marked dirty when we are reactivated, we stomp on the bit that tracks whether we're on the list already. (Also, _dirty's lifecycle is already pretty complicated.)

If we're worried about memory usage, we could use a BitField for storing the bits (active, dirty, this one, and a debug bit).